components/brave_rewards/resources/ui/components/adsBox.tsx

```diff
@@ -111,8 +120,139 @@ class AdsBox extends React.Component<Props, State> {
+        date: item.date,
+        adDetailRows: (
+          item.adDetailRows.map((itemDetail: Rewards.AdHistoryDetail) => {
+            const adContent: Rewards.AdContent = {
+              uuid: itemDetail.adContent.uuid,
+              creativeSetId: itemDetail.adContent.creativeSetId,
+              brand: itemDetail.adContent.brand,
+              brandInfo: itemDetail.adContent.brandInfo,
+              brandLogo: itemDetail.adContent.brandLogo,
+              brandDisplayUrl: itemDetail.adContent.brandDisplayUrl,
+              brandUrl: itemDetail.adContent.brandUrl,
+              likeAction: itemDetail.adContent.likeAction,
+              adAction: itemDetail.adContent.adAction,
+              savedAd: itemDetail.adContent.savedAd,
+              flaggedAd: itemDetail.adContent.flaggedAd,
+              onThumbUpPress: () =>
+                this.onThumbUpPress(itemDetail.adContent.uuid,
+                                    itemDetail.adContent.creativeSetId,
+                                    itemDetail.adContent.likeAction),
+              onThumbDownPress: () =>
+                this.onThumbDownPress(itemDetail.adContent.uuid,
+                                      itemDetail.adContent.creativeSetId,
+                                      itemDetail.adContent.likeAction),
+              onMenuSave: () =>
+                this.onMenuSave(itemDetail.adContent.uuid,
+                                itemDetail.adContent.creativeSetId,
+                                itemDetail.adContent.savedAd),
+              onMenuFlag: () =>
+                this.onMenuFlag(itemDetail.adContent.uuid,
+                                itemDetail.adContent.creativeSetId,
+                                itemDetail.adContent.flaggedAd)
+            }
+            const categoryContent: Rewards.CategoryContent = {
+              category: itemDetail.categoryContent.category,
+              optAction: itemDetail.categoryContent.optAction,
+              onOptInAction: () =>
+                this.onOptInAction(itemDetail.categoryContent.category,
+                                   itemDetail.categoryContent.optAction),
+              onOptOutAction: () =>
+                this.onOptOutAction(itemDetail.categoryContent.category,
+                                    itemDetail.categoryContent.optAction)
+            }
+            return {
+              id: itemDetail.id,
+              adContent: adContent,
+              categoryContent: categoryContent
+            }
+          })
+        )
+      }
+    })
+  }
+
+  hasSavedEntries = (adHistoryData: Rewards.AdsHistoryData[]) => {
+    for (let ix = 0; ix < adHistoryData.length; ix++) {
+      for (let jx = 0; jx < adHistoryData[ix].adDetailRows.length; jx++) {
+        if (adHistoryData[ix].adDetailRows[jx].adContent.savedAd) {
+          return true
+        }
+      }
+    }
+    return false
+  }
+
   render () {
-    // Default values from storage.ts
+    const adsPerHour = 2
```

Comment:
how come we are setting this to 2?

Reply:
@jasonrsadler Wasn't sure on this one...? Is this just a default value?

Reply:
No. This was an example value. Ads per hour should be obtained from whatever the user set in drop down ads setting.

Reply:
OK, will get that in there. Now I remember you mentioned this to me before!